src/types/profile-derived.js

```diff
@@ -97,6 +97,19 @@ export type LineTimings = {|
   selfLineHits: Map<LineNumber, number>,
 |};
 
+// Stores the information that's needed to prove to the symbolication API that
+// we are authorized to request the source code for a specific file.
```

Comment:
Hmm, how would CORS protect against it? You mean the symbol server could send `Access-Control-Allow-Origin: https://profiler.firefox.com` so that only profiler.firefox.com can request source code? But then the server would still need to trust that profiler.firefox.com hasn't been compromised. And it wouldn't provide any protection against non-browser requests.

I'm going to add this comment:

```js
// This "address proof" makes it easy for the browser (or local symbol server)
// to limit file access to only the set of files which are referenced by trusted
// symbol information, without forcing the browser (or local symbol server) to
// build a full list of such files upfront. Building a full list would take a
// long time - up to a minute. Checking individual addresses is much faster.
//
// By allowing access to only the files referenced by symbol information, we
// avoid giving malicious actors the ability to read arbitrary files.
// Specifically, this restriction protects against the following threats:
//  - If source code is requested from the browser via the WebChannel, the
//    browser is protected against a compromised profiler.firefox.com web page
//    or a compromised profiler.firefox.com content process. So in this case the
//    address proof check only makes a difference in cases where the browser can
//    no longer trust the WebChannel.
//  - If source code is requested from a local symbol server via an HTTP request,
//    the server is protected against a compromised profiler.firefox.com page, or
//    against other web pages or outside actors who have guessed the correct URL
//    to request source code from. Symbol servers will usually put a randomized
//    token into the URL in order to make it even harder to guess the right URL.
//    So the address proof check is just an extra layer of protection, in case
//    the secret URL somehow leaks.
```

Reply:
> You mean the symbol server could send Access-Control-Allow-Origin: https://profiler.firefox.com so that only profiler.firefox.com can request source code? 

Yeah that would be impractical indeed, and also then we couldn't use it from deploy previews or local development server.
I'm not too concerned about non-browser requests for local servers that are only listening on localhost :-)

Thanks for the helpful comment!